### DISCUSSION TITLE
STF Service cannot be started after rebooting Android 11 device with PIN protection

### DISCUSSION THREAD

**Discussion by ConfusedMerlin** [DeviceFarmer/stf#545] [Q&A] [answered]
Greetings,

we noticed that our devicefarm cannot reconnect to a device that has been restarted and is now awaiting PIN input. For now, we have only Android 11 devices where the situations "needs restart" and the "is SIM-PIN and/or Device-PIN protected" meets. Which requires somebody to walk up to our fam and input the pin before proceeding.

During this time, the phone usually end up as being stuck in the "preparing" state. And the log is either "error, cannot find/start service" or "waiting for installation to finish...". And while the device is still accessible using adb, installing the STF Service manually does not help either. 

I am aware that this is something which we have to live with, because it's the way Android works. But, does it apply to all supported android versions? 
I suspect that there is somewhere a sentence or two about this topic hidden in your docs, but can't find it. (or, like last time, I just spot it after sending off this discussion). 
Could you please place a statement (or link) which clarifies that?

**Comment by koral--** [comment, marked as answer]
It does not apply to older devices. I'm not sure exactly starting from which Android version it started appearing and whether it applies to all devices or only particular models (especially on the oldest versions). Something about Android 10 AFAIR.

**Reply by ConfusedMerlin** [reply]
Is about what we found out through testing. 
I am about to put a notification Popup behind the reboot-button in the devicefarm, which will tell the user, that his device might not come up by itself if its Android is later than 10. Sadly, my Angular knowledge is still not good enough to limit this check to devices with os greater equal 10.... but this is part of another question.
